business/layer.go

```diff
@@ -102,7 +103,11 @@ func Get(authInfo *api.AuthInfo) (*Layer, error) {
 
 	// Create Jaeger client
 	jaegerLoader := func() (jaeger.ClientInterface, error) {
-		return jaeger.NewClient(authInfo.Token)
+		var err error
+		if jaegerClient == nil {
+			jaegerClient, err = jaeger.NewClient(authInfo.Token)
```

Comment:
Does this token expire? If yes the client will fail to communicate after expiration and new one will not be created?

Reply:
I'm not sure if this feature was ever used.
I've sent a patch to ensure that if the client fails, it's not cached, so in the case of token expiration a following call should populate refresh the client.

In downstream this is not used, as it has an internal user, and in upstream prometheus is not protected with auth as well, so, perhaps it would be needed more work in those cases, but it wouldn't modify the patch.

If reusing the jaeger works well, I think that patch looks good.

Reply:
I will keep it running over night and approve in the morning if it's working fine.